lib/tmpdir.rb

```diff
@@ -13,15 +13,20 @@
 
 class Dir
 
-  @@systmpdir ||= defined?(Etc.systmpdir) ? Etc.systmpdir : '/tmp'
+  # Class variables are inaccessible from non-main Ractor.
+  # And instance variables too, in Ruby 3.0.
+
+  # System-wide temporary directory path
+  SYSTMPDIR = (defined?(Etc.systmpdir) ? Etc.systmpdir.freeze : '/tmp')
```

Comment:
Is it acceptable to freeze `Etc.systemdir` here?  I was surprised that it was not already frozen, but maybe `.dup.freeze` would be better?

Reply:
This method returns a new unfrozen string each time.
We shouldn't freeze the method itself because we can't deny the possibility of code like `Etc.systmpdir << "/foo.txt"`.

Reply:
Ah, I didn't realize `Etc.systmpdir` returns a new string.  This should be fine to merge then.  Thanks!